src/sds_data_model/table.py

```diff
@@ -0,0 +1,204 @@
+from dataclasses import dataclass
+import inspect
+import io
+import json
+from logging import INFO, info, basicConfig
+import requests
+from typing import Any, Dict, List, Optional, TypeVar, Union
+
+from pandas import (DataFrame, Series,
+                    read_csv, read_json, read_excel, read_parquet,
+                    to_numeric)
+from pathlib import Path
+from urllib.parse import urlparse
+
+from sds_data_model.metadata import Metadata
+
+basicConfig(format="%(levelname)s:%(asctime)s:%(message)s", level=INFO)
+
+_TableLayer = TypeVar("_TableLayer", bound="TableLayer")
+
+
+def _find_closing_bracket(string):
+    """Find the index of the closing bracket in a string."""
+    flag = 1
+    string_index = 1
+    while flag > 0:
+        if string[string_index] == "(":
+            flag += 1
+        elif string[string_index] == ")":
+            flag -= 1
+        string_index += 1
+    return string_index - 1
+
+
+def _get_function_input(func_name: str, frame: inspect.types.FrameType) -> str:
+    """Get the input code arguments to a function as a string."""
+    code_input = inspect.getouterframes(frame, 1)
+    code_input = [f for f in code_input
+                  if f'{func_name}(' in f.code_context[0]]
+    code_input = code_input[0].code_context[0]
+    code_input = code_input[len(func_name) + code_input.find(f'{func_name}('):]
+    code_input = code_input[1:_find_closing_bracket(code_input)]
+    return code_input
+
+
+@dataclass
+class TableLayer:
+    """Class for simple rectangular table data.
+
+    Attributes:
+        name (str): Name of data.
+        df (DataFrame): pandas dataframe storing data.
+        metadata (Metadata): metadata for data.
+
+
+    """
+    name: str
+    df: DataFrame
+    metadata: Metadata
+
+    @classmethod
+    def from_file(
+        cls: _TableLayer,
+        data_path: str,
+        data_kwargs: Optional[Dict[str, Any]] = {},
+        metadata_path: Optional[str] = None,
+        name: Optional[str] = None,
+    ) -> _TableLayer:
+        """Load tabular data from a file.
+
+        Args:
+            data_path (:obj:`str`): filepath or url for data.
+            data_kwargs (:obj:`dict`, optional): Additional arguments to pass to data reader.
+            metadata_path (:obj:`str`, optional): filepath or url for metadata.
+            name (:obj:`str`, optional): Name of data.
+
+        Returns:
+            TableLayer object.
+
+        Raises:
+            NotImplementedError
+
+
+    """
+        file_reader = {".csv": read_csv,
+                       ".json": read_json,
+                       ".xlsx": read_excel,
+                       ".xls": read_excel,
```

Comment:
```suggestion
                       ".xls": read_excel,
                       ".xlsm": read_excel,
                       ".xlsb": read_excel,
                       ".odf": read_excel,
                       ".ods": read_excel,
                       ".odt": read_excel, 
```
[pandas.read_excel](https://pandas.pydata.org/docs/reference/api/pandas.read_excel.html) supports several other file types, so we might as well include them too